### PR TITLE
Management API: Property editor temporary file handling

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

This PR applies the new temporary file handling to all applicable property editors.

As part of the implementation, most of (all of?) the old backoffice file handling has been removed - i.e. `ContentPropertyFile`, `IHaveUploadedFiles`, `FileUploadCleanupFilterAttribute`, ...

### Testing this PR

Specifics for each affected property editor are listed below. 

For each test, remember to verify that temporary files are cleaned up automatically after a successful test.

Please note that you should test this _without_ launching the site (`"launchBrowser": false` in `launchSettings.json`) to avoid issues with data type configuration (de)serialization.

#### File Upload

When the submitted value is a valid temporary file key, the temporary file will be saved to the `/media` folder and used as property value.

The former rules for replacing existing values still apply, meaning the following scenarios must also be tested:

1. Update an existing (non-empty) property value. The current file should be deleted from disk afterwards (as the new file takes its place).
2. Submit an empty value for an existing (non-empty) property value. The property value should be cleared, and the current file should be deleted from disk.

#### Image Cropper

Same rules apply here as for File Upload. The only difference is that the file value (i.e. where one would specify the temporary file key) is specified in the `src` part of the property value.

#### Rich Text Editor

The RTE looks for temporary media keys embedded in the markup value using `data-tmpimg` attributes.

Test that when submitting a value that contains ``, a new image created in the media library. When subsequently fetching the value again, the `data-tmpimg` will be replaced with a `data-udi` attribute containing the newly created media UDI.

Also test that:
1. Submitting a value with an existing `data-udi` attribute yields no changes in the media library.
2. Removing an existing `data-udi` attribute does _not_ cause the corresponding media item to be deleted in the media library.

#### Media Picker

The Media Picker is the only one of these property editors that require an authorized backoffice user (the others perform fallback to super user). Since auth isn't enabled in the new backoffice yet, it's necessary to hardcode the super user key in the Media Picker value editor [here](https://github.com/umbraco/Umbraco-CMS/pull/14244/files#diff-015c5fbe7934addfa41d450dc4fe855bbf21b0fc669a564f1621b1505949f900R222).

In the current backoffice, the Media Picker allows for picking the desired media type, if multiple media types can be applied to the new file. However, the media type sent to the server is not modelled in any request models... it is conveniently wrapped and sent in an anonymous model and detected at server level using JSON parsing. Besides being very implicit, this also won't work for the management API, so I have added the media type to the Media Picker models for both requests and responses. In other words, the media type is required when testing Media Picker.

To apply a temporary file to a Media Picker, replace the `mediaKey` value with a valid temporary file key.

### CODE REVIEW

File: src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs [hunk 1]
```diff
@@ -188,48 +204,63 @@ internal static IEnumerable<MediaWithCropsDto> Deserialize(IJsonSerializer jsonS
...
+            foreach (MediaWithCropsDto mediaWithCropsDto in mediaWithCropsDtos)
+            {
+                IMedia? media = _mediaService.GetById(mediaWithCropsDto.MediaKey);
+                mediaWithCropsDto.MediaTypeAlias = media?.ContentType.Alias ?? unknownMediaType;
+            }
+
+            return mediaWithCropsDtos.Where(m => m.MediaTypeAlias != unknownMediaType).ToList();
+        }
+
+        List<MediaWithCropsDto> HandleTemporaryMediaUploads(List<MediaWithCropsDto> mediaWithCropsDtos, MediaPicker3Configuration configuration)
```
Comment (nikolajlauridsen) [inline src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs:220]:
```suggestion
        private List<MediaWithCropsDto> HandleTemporaryMediaUploads(List<MediaWithCropsDto> mediaWithCropsDtos, MediaPicker3Configuration configuration)
```

I'm assuming we want this to be private 😄

**Review by nikolajlauridsen — COMMENTED** [review]
Looks great to me, only found 2 very minor things 🎉

File: src/Umbraco.Core/Services/MediaImportService.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public async Task<IMedia> ImportAsync(string fileName, Stream fileStream, Guid? parentId, string? mediaTypeAlias, Guid userKey)
+    {
+        if (fileStream.CanRead == false)
+        {
+            throw new InvalidOperationException("Could not read from file stream, please ensure it is open and readable");
+        }
+
+        var userId = await _userIdKeyResolver.GetAsync(userKey);
+        IUser user = _userService.GetUserById(userId)
+                     ?? throw new ArgumentException($"Could not find a user with the specified user key ({userKey})", nameof(userKey));
```
[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Core/Services/MediaImportService.cs:54]: Ooh of course 🤦 makes great sense; we can even save a dependency on the `IUserIdKeyResolver` then! I'll update :)

[the remark this answers is not in the document]
Comment (kjac) [reply src/Umbraco.Core/Services/MediaImportService.cs]: Fixed 👍

[the remark this answers is not in the document]
Comment (nikolajlauridsen) [reply src/Umbraco.Core/Services/MediaImportService.cs]: Oh that's true, and great, love it 🎉

**Pull request merged**